text/0039-service-overhaul.md

```diff
@@ -0,0 +1,196 @@
+
+### New service JSON format
+
+**Needs RFC 38 and resolutions matrix ajustments**
+
+Here is a example:
+```json
+{
+    "format_version": 4,
+    "services": [
+        {
+            "id": "example",
+            "name": "Example of stream services",
+            "more_info_link": "https://example.com/more_info",
+            "stream_key_link": "https://example.com/stream_key",
+            "available_protocols": ["RTMP","FTL"],
+            "servers": [
+                {
+                    "name": "Example RTMP 1",
+                    "url": "rtmp://example.com/server1"
+                },
+                {
+                    "protocol": "RTMP",
+                    "name": "Example RTMP 2",
+                    "url": "rtmp://example.com/server2"
+                },
+                {
+                    "protocol": "FTL",
+                    "name": "Example FTL",
+                    "url": "ftl://example.com/server_ftl"
+                }
+            ],
+            "maximum": [
+                {
+                    "protocol": "RTMP",
+                    "video_bitrate": 3000,
+                    "audio_bitrate": 320
+                },
+                {
+                    "protocol": "FTL",
+                    "fps": 30,
+                }
+            ],
+            "supported_resolutions": [
+                "1920x1080",
+                "1280x720",
+                "852x480",
+                "480x360"
+            ],
+            "recommended": [
+                {
+                    "protocol": "RTMP"
+                    // Work In Progress
+                }
+            ]
+        }
+    ]
+}
+```
+
+In this new format:
+- Each service will also have his own id and be registered with it.
+- Link for more info and stream key can be added.
+- Multiple protocols can be set for one service, if not it will default to RTMP only.
+- Servers have now the attribute "protocol", if not provided it will default to RTMP.
+- Maximums are now separated from recommended settings and can be set for each protocol but no settings for any protocol. **And the protocol shall be set, there will be no fallback to RTMP**
+- Supported resolutions are separated to, it's usualy because of the player used by the service, so it's set for any protocol. **Need to add new matrix logic**
+- Recommended settings and can be set for each protocol but no settings for any protocol. **And the protocol shall be set, there will be no fallback to RTMP.** *Consider it as futureproofing for when new codec for other protocol like AV1 will come. RTMP spec tell that it only support H264.*
+
+### Service ID naming scheme
+- Only lower case letter
+- Hyphen `-` are only used to distinguish two service providing the same stream service but with noteworthy differences like NicoNico (free & premium) or Twitch and Twitch OAuth
+
+### VOD Track
+This option will only be shown if the combo service-protocol is compatible.
```

Comment:
This probably needs additional work not specified here other than "just a checkbox"

Reply:
I don't understand what you mean with the "VOD Track" and a "just a checkbox" ?

Reply:
Audio track capability was added to the Services API.